### DISCUSSION TITLE
Problem with characteristicShape function

### DISCUSSION THREAD

**Discussion by JasonOttUW** [mtex-toolbox/mtex#1649] [Ask Anything] [answered]
I was looking at projection based shape parameters for one of my maps, and was trying to use the 'characteristicShape' function as discussed on the page: https://mtex-toolbox.github.io/ProjectionBasedParameters.html
applying this to the phases in my sample, I entered the command as: 
shape = characteristicShape(grains.boundary('gl','gl'))
and got back an error that I was not using enough input arguments.
I checked the documentation for the function, and it is expecting input of:
[csAngle, csRadius] = characteristicShape(bc,bp)
where bc is the bin center and bp is the bin population. So, I am quite confused about this. Can anyone clear up what the correct usage is for this function if I want to get the average grain shape for this phase?
Thanks in advance!

**Comment by kilir** [comment, marked as answer]
Hi Jason, 
the documented syntax is rather new while the one you get at your installation is the older one, which required a different input (obtained from frunction called calcTDF).
Hope that helps.
Cheers,
Rüdiger

**Reply by JasonOttUW** [reply]
Thank you-got the new version and the problem was solved!
